### DISCUSSION TITLE
Error when attempting logging during Renderer crash

### DISCUSSION THREAD

**Discussion by douglassllc** [megahertz/electron-log#348] [Q&A] [answered]
When a renderer process crashed I receive the following error when attempting to use the log functions within my event handlers that catch the crash.

Example code
```
app.on('render-process-gone', async function (event, webContents, details) {
  log.info('Renderer process has crashed!')
})
```

Error thrown
```
Error: Render frame was disposed before WebFrameMain could be accessed
    at maxDepth (object.js:86:31)
    at maxDepth (object.js:86:18)
    at object.js:62:14
    at Array.map (<anonymous>)
    at maxDepth (object.js:61:17)
    at maxDepthFunction (object.js:96:12)
    at index.js:47:14
    at Array.reduce (<anonymous>)
    at Object.transform (index.js:45:23)
    at transformMain (console.js:59:20)
```

If the Renderer is crashing should I still be able to use the `log.*` functions?

### Versions
"electron-log": "^4.4.8"
"electron": "^20.3.8"

**Comment by megahertz** [comment]
Looks like it tries to serialize some complex object. In such a case, the logger still works fine in the main process. Anyway, this behavior is changed in v5 beta.

**Comment by douglassllc** [comment, marked as answer]
@megahertz The call that is throwing the error is in `electronApi.js` file.

```
function sendIpc(channel, message) {
  if (process.type === 'browser') {
    sendIpcToRenderer(channel, message);  // <-- THIS LINE IS ATTEMPTING TO SEND TO THE CRASHED RENDERER
  } else if (process.type === 'renderer') {
    sendIpcToMain(channel, message);
  }
}
```

Should I attempt to update to v5 beta or is their away in my current version to disable attempts to log message in the renderer if the renderer is crashing?

I was thinking that maybe I should be disabling the console transport when the renderer crashes and re-enable it when the renderer has been respawned.  

Also of note is that I have only been working in DEV mode.  I don't know if this same issue would happen with my PROD build.

**Reply by megahertz** [reply]
In v4 you could disabled ipc transport for both main and renderer processes. It's only necessary for debugging and is disabled by default for a packaged app.

**Reply by douglassllc** [reply]
@megahertz Thank you.  Your recommendation worked.
